### PR TITLE
dht: couple Get in CLI with DHT and implement findValueRequestHandler

### PR DESCRIPTION
**Pull request by optmzr**
* ctl: prettify and fix RPC gob encoding errors
 * cmd: change -exit to bool flag
 * store: make item.value -> item.Value (public)

Fixes: #53

### CODE REVIEW

File: cmd/dhtctl/main.go [hunk 1]
```diff
@@ -11,67 +11,60 @@ import (
...
-	if reply {
-		fmt.Println("RPC successful")
-	}
+	log.Printf("Value: %s\n", value)
 }
 
 func ping(c *rpc.Client, id node.ID) {
 	ping := ctl.Ping{NodeID: id}
-	var reply []byte
+	var challenge []byte
```
Comment (edvinnn) [inline cmd/dhtctl/main.go]: You don't like my variable names? :'(

File: cmd/dhtctl/main.go [hunk 2]
```diff
@@ -11,67 +11,60 @@ import (
...
+		log.Fatalln("Ping error:", err)
 	}
 
-	fmt.Println("Ping response: ", reply)
+	fmt.Printf("Ping response: %x\n", challenge)
 }
 
-func exit(c *rpc.Client, id node.ID) {
-	var reply bool
+func exit(c *rpc.Client) {
```
Comment (edvinnn) [inline cmd/dhtctl/main.go]: Should'n we take a nodeID as an argument? What if we want to terminate another node?

File: dht/dht.go [hunk 3]
```diff
@@ -288,32 +321,52 @@ func (dht *DHT) iterativeStore(value string) (hash store.Key, err error) {
...
 
 	return
 }
 
 func (dht *DHT) iterativeFindValue(hash store.Key) (value string, err error) {
 	call := NewFindValueCall(hash)
-	if _, err = dht.walk(call); err == nil {
+	closest, err := dht.walk(call)
+	if err != nil {
+		return
```
Comment (edvinnn) [inline dht/dht.go]: Assuming this is right? 🙄

File: dht/dht.go [hunk 4]
```diff
@@ -288,32 +321,52 @@ func (dht *DHT) iterativeStore(value string) (hash store.Key, err error) {
...
+	}
+
+	if call.value != "" {
 		value = call.value
+	} else {
+		err = fmt.Errorf("Couldn't find any value with the hash: %v", hash)
+		return
 	}
+
+	// Store at the closest node that did not return any value.
```
Comment (edvinnn) [inline dht/dht.go]: Is this part of Kademlia? 🙄 😮

File: cmd/dhtctl/main.go (hunk 2 above)
Comment (optmzr) [reply cmd/dhtctl/main.go]: The lab spec. says "the node", so I thought that we'd skip that feature :)

File: dht/dht.go (hunk 3 above)
Comment (optmzr) [reply dht/dht.go]: Yes, it'll return the error

File: dht/dht.go (hunk 4 above)
Comment (optmzr) [reply dht/dht.go]: Yep
2019-09-29-195202_845x56_scrot

File: cmd/dhtctl/main.go (hunk 1 above)
Comment (optmzr) [reply cmd/dhtctl/main.go]: sorry :sob:

File: dht/dht.go (hunk 3 above)
Comment (edvinnn) [reply dht/dht.go]: I'm a noob obviously 🙃 🙅‍♂

**Comment by optmzr** [comment]
Only *DECENT*?!?!?!!? Should I not merge?!

**Comment by optmzr** [comment]
Missing meme, too, blocked.

**Pull request merged**